client/modules/core/main.js

```diff
@@ -158,6 +159,46 @@ export default {
     return reactionState;
   },
 
+  /**
+   * absoluteUrl
+   * @summary a wrapper method for Meteor.absoluteUrl which sets the rootUrl to
+   * the current URL (instead of defaulting to ROOT_URL)
+   * @param {String} [path] A path to append to the root URL. Do not include a leading "`/`".
+   * @param {Object} [options]
+   * @param {Boolean} options.secure Create an HTTPS URL.
+   * @param {Boolean} options.replaceLocalhost Replace localhost with 127.0.0.1. Useful for services that don't recognize localhost as a domain name.
+   * @param {String} options.rootUrl Override the default ROOT_URL from the server environment. For example: "`http://foo.example.com`"
+   * @return {String} URL for the given path and options
+   */
+  absoluteUrl(path, options) {
+    const o = _.extend({}, options);
+    const domain = this._shopDomain.get();
+
+    if (!("rootUrl" in o) && domain) {
+      // unfortunately, the scheme/protocal is not available here, so let's
+      // get it from the default Meteor absoluteUrl options, then replace the
+      // host
+      const rootUrl = url.parse(Meteor.absoluteUrl.defaultOptions.rootUrl);
+      rootUrl.host = domain;
+      o.rootUrl = rootUrl.format();
+    }
+
+    return Meteor.absoluteUrl(path, o);
+  },
+
+  /**
+   * getDomain
+   * @summary extracts the domain name from the absoluteUrl.
+   * @param {Object} [options] inheritied from absoluteUrl
+   * @param {Boolean} options.secure Create an HTTPS URL.
+   * @param {Boolean} options.replaceLocalhost Replace localhost with 127.0.0.1. Useful for services that don't recognize localhost as a domain name.
+   * @param {String} options.rootUrl Override the default ROOT_URL from the server environment. For example: "`http://foo.example.com`"
+   * @return {String} Domain/hostname for the given options
+   */
+  getDomain(options) {
+    return this.absoluteUrl("", options).split("/")[2].split(":")[0];
```

Comment:
This is copied it from [another part of the code-base](https://github.com/reactioncommerce/reaction/blob/b297619593a15dc0fd751c343970cd53ae89c42b/client/modules/core/main.js#L97) to keep a single technique for extracting the domain.

Reply:
for sure. I didn't want to create a second pattern for doing the same thing, so I copy/pasted this logic.
I will make this `url.parse` and attempt to update the client-side test as well, assuming the URL library is available there.